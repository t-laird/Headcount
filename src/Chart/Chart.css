.Chart {
  position: absolute;
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  justify-content: space-between;
  top: 220px;
  left: calc(50% - 300px);
  height: 380px;
  width: 600px;
  background-color: white;
  border: 1px solid black; }
  .Chart .chartNav {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    flex-wrap: nowrap;
    width: 100%;
    height: 40px;
    box-sizing: border-box; }
    .Chart .chartNav h5 {
      padding-left: 8px;
      margin: 0px;
      margin-top: 10px; }
    .Chart .chartNav .district1key {
      background-color: red;
      border-radius: 5px;
      height: 10px;
      width: 10px;
      display: inline-block;
      margin-left: 10px; }
    .Chart .chartNav .district2key {
      background-color: blue;
      border-radius: 5px;
      height: 10px;
      width: 10px;
      display: inline-block;
      margin-left: 10px; }
  .Chart ul {
    width: 100%;
    display: flex;
    padding: 0px;
    justify-content: space-between; }
    .Chart ul li {
      display: inline-block;
      text-align: center; }
  .Chart svg {
    position: absolute;
    left: calc(50% - 300px);
    width: 600px;
    height: 300px;
    padding-top: 40px; }
