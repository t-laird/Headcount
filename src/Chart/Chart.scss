.Chart {
  position: absolute;
  display: flex;
  flex-wrap: wrap;
  flex-direction: column;
  justify-content: space-between;
  top: 180px;
  left: calc(50% - 160px);
  height: 380px;
  width: 600px;
  background-color: white;
  border: 1px solid black;
  .chartNav {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    flex-wrap: nowrap;
    width: 100%;
    height: 40px;
    box-sizing: border-box;
    h5 {
      padding-left: 8px;
      margin: 0px;
      margin-top: 10px;
    }
    .district1key {
        background-color: red;
        border-radius: 5px;
        height: 10px;
        width: 10px;
        display: inline-block;
        margin-left: 10px;
    }
    .district2key {
      background-color: blue;
      border-radius: 5px;      
      height: 10px;
      width: 10px;
      display: inline-block;
      margin-left: 10px;          
    }
    button {
      z-index: 10;
      border-radius: 0px;
      border: none;
      background: lighten(red, 15%);
      width: 100px;
      margin-left: 100px;
      color: #fff;
      font-size: 14pt;
      cursor: pointer;
      &:hover {
        background: #506479;
      }
    }
  }
  ul {
    width: 100%;
    display: flex;
    padding: 0px;
    justify-content: space-between;
    li {
      display: inline-block;
      text-align: center;
    }
  }
  svg {
      position: absolute;
      left: calc(50% - 300px);
      width: 600px;
      height: 300px;
      padding-top: 40px;
  }
}