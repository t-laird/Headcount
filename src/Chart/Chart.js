import React from 'react';
import {Sparklines, SparklinesLine, SparklinesReferenceLine } from 'react-sparklines';
import './Chart.css';
import Button from '../Button/Button';

const Chart = (props) => {
  const comparedLocations = props.cards.filter( card => {
    return Object.keys(card)[0] === props.comparisons[0] || Object.keys(card)[0] === props.comparisons[1]
  });

  const location1 = Object.assign({}, 
                      {district: Object.keys(comparedLocations[0])[0]}, 
                      {scores: Object.values(comparedLocations[0][Object.keys(comparedLocations[0])[0]])}, 
                      {dates: Object.keys(comparedLocations[0][Object.keys(comparedLocations[0])[0]])}
                    );

  const location2 = Object.assign({}, 
                      {district: Object.keys(comparedLocations[1])[0]}, 
                      {scores: Object.values(comparedLocations[1][Object.keys(comparedLocations[1])[0]])}, 
                      {dates: Object.keys(comparedLocations[1][Object.keys(comparedLocations[1])[0]])}
                    );


  const mappedDates = location1.dates.map( (date, index) => {
    return <li style={{width: 600/location1.dates.length + 'px'}} key={index}>{date}</li>;
  });

  return (
    <div className="Chart">
      <Sparklines height={50} width={100} max={1.0} min={0} data={location1.scores}>
        <SparklinesLine color="red" style={{fill:'none'}}/>
      </Sparklines>
      <Sparklines height={50} width={100} max={1.0} min={0} data={location2.scores}>
        <SparklinesLine color="blue" style={{fill:'none'}}/>
      </Sparklines>
      <div className="chartNav">
        <h5>{location1.district}<div className="district1key"></div></h5>
        <h5>{location2.district}<div className="district2key"></div></h5>
        <Button type="Close"/>
      </div>
      <ul>
        {mappedDates}
      </ul>
    </div>
  );
}

export default Chart;