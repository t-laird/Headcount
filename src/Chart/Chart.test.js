/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import Chart from './Chart';
import { shallow, mount } from 'enzyme';

describe('Chart test', () => {
  let mockFunc;
  let renderedChart;
  let mockData;

  beforeEach(() => {
    mockFunc = jest.fn();
    mockData = [{'Colorado': {2005: 0.351, 2006: 0.619, 2007: 0.814}}, {'ACADEMY 20': {2005: 0.123, 2006: 0.234, 2007: 0.345}}];
    renderedChart = shallow(<Chart comparisons={['Colorado', 'ACADEMY 20']} chartStatus={mockFunc} cards={mockData} />);
  });

  it('should render the correct chart elements', () => {
    expect(renderedChart.find('Sparklines').length).toEqual(2);
    expect(renderedChart.find('ul').length).toEqual(1);
    expect(renderedChart.find('li').length).toEqual(3)
    expect(renderedChart.find('Button').length).toEqual(1);
    expect(renderedChart.find('.axis').length).toEqual(1);
    expect(renderedChart.find('span').length).toEqual(2);
  });
  
  it('should render the correct number of li components for the x-axis given more or less data', () => {
    const largerMockData = [{'Colorado': {2005: 0.351, 2006: 0.619, 2007: 0.814, 2008: 0.999}}, {'ACADEMY 20': {2005: 0.123, 2006: 0.234, 2007: 0.345, 2008: 0.955}}];
    renderedChart = shallow(<Chart comparisons={['Colorado', 'ACADEMY 20']} chartStatus={mockFunc} cards={largerMockData} />);
    expect(renderedChart.find('li').length).toEqual(4);
  });

  it('Should match the snapshot', () => {
    expect(renderedChart).toMatchSnapshot();
  });
  
  it('should adjust the y-axis to accomodate larger data when necessary', () => {
    const largerMockData = [{'Colorado': {2005: 351, 2006: 619, 2007: 814, 2008: 999}}, {'ACADEMY 20': {2005: 123, 2006: 234, 2007: 345, 2008: 955}}];
    renderedChart = shallow(<Chart comparisons={['Colorado', 'ACADEMY 20']} chartStatus={mockFunc} cards={largerMockData} />);
    
    expect(renderedChart.find('span').first().text()).toEqual("0");
    expect(renderedChart.find('span').last().text()).toEqual("999");
  });
});