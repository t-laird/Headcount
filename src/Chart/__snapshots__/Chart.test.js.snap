exports[`Chart test Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Chart
    cards={
        Array [
            Object {
              "Colorado": Object {
                "2005": 0.351,
                "2006": 0.619,
                "2007": 0.814,
              },
            },
            Object {
              "ACADEMY 20": Object {
                "2005": 0.123,
                "2006": 0.234,
                "2007": 0.345,
              },
            },
          ]
    }
    chartStatus={[Function]}
    comparisons={
        Array [
            "Colorado",
            "ACADEMY 20",
          ]
    } />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="chart-container">
          <div
                    className="axis">
                    <span>
                              0
                    </span>
                    <span>
                              1
                    </span>
          </div>
          <Sparklines
                    data={
                              Array [
                                        0.351,
                                        0.619,
                                        0.814,
                                      ]
                    }
                    height={50}
                    margin={2}
                    max={1}
                    min={0}
                    preserveAspectRatio="none"
                    width={100}>
                    <SparklinesLine
                              color="red"
                              onMouseMove={[Function]}
                              style={
                                        Object {
                                                  "fill": "none",
                                                }
                              } />
          </Sparklines>
          <Sparklines
                    data={
                              Array [
                                        0.123,
                                        0.234,
                                        0.345,
                                      ]
                    }
                    height={50}
                    margin={2}
                    max={1}
                    min={0}
                    preserveAspectRatio="none"
                    width={100}>
                    <SparklinesLine
                              color="blue"
                              onMouseMove={[Function]}
                              style={
                                        Object {
                                                  "fill": "none",
                                                }
                              } />
          </Sparklines>
</div>,
        <div
          className="chartNav">
          <h5>
                    Colorado
                    <div
                              className="district1key" />
          </h5>
          <h5>
                    ACADEMY 20
                    <div
                              className="district2key" />
          </h5>
          <Button
                    chartStatus={[Function]}
                    name="Close"
                    type="chart-close" />
</div>,
        <ul>
          <li
                    style={
                              Object {
                                        "width": "200px",
                                      }
                    }>
                    2005
          </li>
          <li
                    style={
                              Object {
                                        "width": "200px",
                                      }
                    }>
                    2006
          </li>
          <li
                    style={
                              Object {
                                        "width": "200px",
                                      }
                    }>
                    2007
          </li>
</ul>,
      ],
      "className": "Chart",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="axis">
              <span>
                            0
              </span>
              <span>
                            1
              </span>
</div>,
            <Sparklines
              data={
                            Array [
                                          0.351,
                                          0.619,
                                          0.814,
                                        ]
              }
              height={50}
              margin={2}
              max={1}
              min={0}
              preserveAspectRatio="none"
              width={100}>
              <SparklinesLine
                            color="red"
                            onMouseMove={[Function]}
                            style={
                                          Object {
                                                        "fill": "none",
                                                      }
                            } />
</Sparklines>,
            <Sparklines
              data={
                            Array [
                                          0.123,
                                          0.234,
                                          0.345,
                                        ]
              }
              height={50}
              margin={2}
              max={1}
              min={0}
              preserveAspectRatio="none"
              width={100}>
              <SparklinesLine
                            color="blue"
                            onMouseMove={[Function]}
                            style={
                                          Object {
                                                        "fill": "none",
                                                      }
                            } />
</Sparklines>,
          ],
          "className": "chart-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  0
</span>,
                <span>
                  1
</span>,
              ],
              "className": "axis",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "0",
                },
                "ref": null,
                "rendered": "0",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 1,
                },
                "ref": null,
                "rendered": 1,
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <SparklinesLine
                color="red"
                onMouseMove={[Function]}
                style={
                                Object {
                                                "fill": "none",
                                              }
                } />,
              "data": Array [
                0.351,
                0.619,
                0.814,
              ],
              "height": 50,
              "margin": 2,
              "max": 1,
              "min": 0,
              "preserveAspectRatio": "none",
              "width": 100,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "color": "red",
                "onMouseMove": [Function],
                "style": Object {
                  "fill": "none",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <SparklinesLine
                color="blue"
                onMouseMove={[Function]}
                style={
                                Object {
                                                "fill": "none",
                                              }
                } />,
              "data": Array [
                0.123,
                0.234,
                0.345,
              ],
              "height": 50,
              "margin": 2,
              "max": 1,
              "min": 0,
              "preserveAspectRatio": "none",
              "width": 100,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "color": "blue",
                "onMouseMove": [Function],
                "style": Object {
                  "fill": "none",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h5>
              Colorado
              <div
                            className="district1key" />
</h5>,
            <h5>
              ACADEMY 20
              <div
                            className="district2key" />
</h5>,
            <Button
              chartStatus={[Function]}
              name="Close"
              type="chart-close" />,
          ],
          "className": "chartNav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Colorado",
                <div
                  className="district1key" />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Colorado",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "district1key",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "ACADEMY 20",
                <div
                  className="district2key" />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "ACADEMY 20",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "district2key",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "chartStatus": [Function],
              "name": "Close",
              "type": "chart-close",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              style={
                            Object {
                                          "width": "200px",
                                        }
              }>
              2005
</li>,
            <li
              style={
                            Object {
                                          "width": "200px",
                                        }
              }>
              2006
</li>,
            <li
              style={
                            Object {
                                          "width": "200px",
                                        }
              }>
              2007
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "2005",
              "style": Object {
                "width": "200px",
              },
            },
            "ref": null,
            "rendered": "2005",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "2006",
              "style": Object {
                "width": "200px",
              },
            },
            "ref": null,
            "rendered": "2006",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": "2007",
              "style": Object {
                "width": "200px",
              },
            },
            "ref": null,
            "rendered": "2007",
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="chart-container">
            <div
                        className="axis">
                        <span>
                                    0
                        </span>
                        <span>
                                    1
                        </span>
            </div>
            <Sparklines
                        data={
                                    Array [
                                                0.351,
                                                0.619,
                                                0.814,
                                              ]
                        }
                        height={50}
                        margin={2}
                        max={1}
                        min={0}
                        preserveAspectRatio="none"
                        width={100}>
                        <SparklinesLine
                                    color="red"
                                    onMouseMove={[Function]}
                                    style={
                                                Object {
                                                            "fill": "none",
                                                          }
                                    } />
            </Sparklines>
            <Sparklines
                        data={
                                    Array [
                                                0.123,
                                                0.234,
                                                0.345,
                                              ]
                        }
                        height={50}
                        margin={2}
                        max={1}
                        min={0}
                        preserveAspectRatio="none"
                        width={100}>
                        <SparklinesLine
                                    color="blue"
                                    onMouseMove={[Function]}
                                    style={
                                                Object {
                                                            "fill": "none",
                                                          }
                                    } />
            </Sparklines>
</div>,
          <div
            className="chartNav">
            <h5>
                        Colorado
                        <div
                                    className="district1key" />
            </h5>
            <h5>
                        ACADEMY 20
                        <div
                                    className="district2key" />
            </h5>
            <Button
                        chartStatus={[Function]}
                        name="Close"
                        type="chart-close" />
</div>,
          <ul>
            <li
                        style={
                                    Object {
                                                "width": "200px",
                                              }
                        }>
                        2005
            </li>
            <li
                        style={
                                    Object {
                                                "width": "200px",
                                              }
                        }>
                        2006
            </li>
            <li
                        style={
                                    Object {
                                                "width": "200px",
                                              }
                        }>
                        2007
            </li>
</ul>,
        ],
        "className": "Chart",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="axis">
                <span>
                                0
                </span>
                <span>
                                1
                </span>
</div>,
              <Sparklines
                data={
                                Array [
                                                0.351,
                                                0.619,
                                                0.814,
                                              ]
                }
                height={50}
                margin={2}
                max={1}
                min={0}
                preserveAspectRatio="none"
                width={100}>
                <SparklinesLine
                                color="red"
                                onMouseMove={[Function]}
                                style={
                                                Object {
                                                                "fill": "none",
                                                              }
                                } />
</Sparklines>,
              <Sparklines
                data={
                                Array [
                                                0.123,
                                                0.234,
                                                0.345,
                                              ]
                }
                height={50}
                margin={2}
                max={1}
                min={0}
                preserveAspectRatio="none"
                width={100}>
                <SparklinesLine
                                color="blue"
                                onMouseMove={[Function]}
                                style={
                                                Object {
                                                                "fill": "none",
                                                              }
                                } />
</Sparklines>,
            ],
            "className": "chart-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    0
</span>,
                  <span>
                    1
</span>,
                ],
                "className": "axis",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "0",
                  },
                  "ref": null,
                  "rendered": "0",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 1,
                  },
                  "ref": null,
                  "rendered": 1,
                  "type": "span",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <SparklinesLine
                  color="red"
                  onMouseMove={[Function]}
                  style={
                                    Object {
                                                      "fill": "none",
                                                    }
                  } />,
                "data": Array [
                  0.351,
                  0.619,
                  0.814,
                ],
                "height": 50,
                "margin": 2,
                "max": 1,
                "min": 0,
                "preserveAspectRatio": "none",
                "width": 100,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "color": "red",
                  "onMouseMove": [Function],
                  "style": Object {
                    "fill": "none",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <SparklinesLine
                  color="blue"
                  onMouseMove={[Function]}
                  style={
                                    Object {
                                                      "fill": "none",
                                                    }
                  } />,
                "data": Array [
                  0.123,
                  0.234,
                  0.345,
                ],
                "height": 50,
                "margin": 2,
                "max": 1,
                "min": 0,
                "preserveAspectRatio": "none",
                "width": 100,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "color": "blue",
                  "onMouseMove": [Function],
                  "style": Object {
                    "fill": "none",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h5>
                Colorado
                <div
                                className="district1key" />
</h5>,
              <h5>
                ACADEMY 20
                <div
                                className="district2key" />
</h5>,
              <Button
                chartStatus={[Function]}
                name="Close"
                type="chart-close" />,
            ],
            "className": "chartNav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Colorado",
                  <div
                    className="district1key" />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Colorado",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "district1key",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
              ],
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "ACADEMY 20",
                  <div
                    className="district2key" />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "ACADEMY 20",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "district2key",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
              ],
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "chartStatus": [Function],
                "name": "Close",
                "type": "chart-close",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                style={
                                Object {
                                                "width": "200px",
                                              }
                }>
                2005
</li>,
              <li
                style={
                                Object {
                                                "width": "200px",
                                              }
                }>
                2006
</li>,
              <li
                style={
                                Object {
                                                "width": "200px",
                                              }
                }>
                2007
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": "2005",
                "style": Object {
                  "width": "200px",
                },
              },
              "ref": null,
              "rendered": "2005",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "2006",
                "style": Object {
                  "width": "200px",
                },
              },
              "ref": null,
              "rendered": "2006",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": "2007",
                "style": Object {
                  "width": "200px",
                },
              },
              "ref": null,
              "rendered": "2007",
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
